3-extra/1-count-words.js

```diff
@@ -27,10 +27,29 @@ function countWords(string) {
   const wordCount = {};
 
   // write code here
+  if (!string) {
+    return wordCount;
+  } else {
+    const split = string.split(" ");
+
+    for (let word of split) {
+      wordCount[word] = count(word, split);
+    }
+  }
 
   return wordCount;
 }
 
+function count(word, splitString) {
+  let count = 0;
+
+  for (let word2 of splitString) {
+    if (word === word2) count++;
+  }
+
+  return count;
+}
+
 /* ======= TESTS - DO NOT MODIFY =====
```

Comment:
There's another way to do it.
```javascript
function countWords(string) {
  const wordCount = {};

  string.split(/\W+/).forEach((word) => {
    if (wordCount[word]) {
      wordCount[word] += 1;
    } else {
      wordCount[word] = 1;
    }
  });

  return wordCount;
}
```
You'd be surprised how often you need to use this technique to process a list and produce a tally like this. Let me know if you understand this. If the word is already a key on the object, incremement a counter. If it's not already a key on the object, set the key to a count of 1 (one word counted).